driver/driver.go

```
@@ -4,69 +4,92 @@
 package driver
 
 import (
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+	"time"
 
+	"context"
+
+	"github.com/Microsoft/go-winio"
 	"github.com/Microsoft/hcsshim"
 	log "github.com/Sirupsen/logrus"
 	"github.com/codilime/contrail-windows-docker/common"
 	"github.com/codilime/contrail-windows-docker/controller"
 	"github.com/codilime/contrail-windows-docker/hns"
+	"github.com/codilime/contrail-windows-docker/hnsManager"
+	dockerClient "github.com/docker/docker/client"
 	"github.com/docker/go-plugins-helpers/network"
-	"github.com/docker/go-plugins-helpers/sdk"
 )
 
 type ContrailDriver struct {
-	controller *controller.Controller
-	HnsID      string
+	controller     *controller.Controller
+	hnsMgr         *hnsManager.HNSManager
+	networkAdapter string
+	listener       net.Listener
 }
 
-func NewDriver(subnet, gateway, adapter string, c *controller.Controller) (*ContrailDriver,
-	error) {
+func NewDriver(adapter string, c *controller.Controller) (*ContrailDriver, error) {
```

Error is never returned. Maybe remove it from prototype?